pkg/ipamd/datastore/data_store.go

```diff
@@ -432,50 +497,52 @@ func (ds *DataStore) DelIPv4AddressFromStore(eniID string, ipv4 string, force bo
 	// Prometheus gauge
 	totalIPs.Set(float64(ds.total))
 
-	delete(curENI.IPv4Addresses, ipv4)
+	curENI.removeAddress(idx)
```

Comment:
Hmm.. So calling this with either the IPv4 or the IPv6 address will remove both from the datastore.